### PR TITLE
[SYCLomatic]Add migration rules and enable lit tests for 4 cmake syntax migration

### PR DESCRIPTION
**Pull request by tomflinda**
This PR coveres 4 cmake syntaxes as below:
1.user defined CUDA specific variables migration
2.target_compile_options
3.add_compile_options
4.CUDA_HAS_FP16

 Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

**Review by AndyCHHuang — APPROVED** [review]
LGTM

**Review by zhiweij1 — APPROVED** [review]

File: clang/test/dpct/cmake_migation/case_021/expected.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+add_compile_options(-std=c++17)
+add_compile_options(-std=c++17)
+add_compile_options(-std=c++17)
+#Currently we use Yaml based migartion rule to migarte -std=c++22 down to -std=c++17.
+#We will implement an implicit migration rule to fix this issue in future.
+add_compile_options(-std=c++17)
```
Comment (zhimingwang36) [inline clang/test/dpct/cmake_migation/case_021/expected.txt]: L6 is not correct,  Please comment out, and uncomment it when it is ready.

File: clang/tools/dpct/cmake/dpct.cmake [hunk 2]
```diff
@@ -74,3 +74,6 @@ endmacro()
 macro(DPCT_COMPILE_SYCL_CODE generated_files)
   DPCT_COMPILE_SYCL_CODE_IMP(sycl_device ${generated_files} ${ARGN})
 endmacro()
+
+# Always set SYCL_HAS_FP16 to true to assume SYCL device to support float16
```
Comment (zhimingwang36) [inline clang/tools/dpct/cmake/dpct.cmake:78]: please emit a message like:  "dpct.cmake: SYCL_HAS_FP16 is set true by default."

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**